ndc-reference/bin/reference/main.rs

```diff
@@ -1477,6 +1505,81 @@ fn eval_comparison_value(
     }
 }
 // ANCHOR_END: eval_comparison_value
+// ANCHOR: eval_nested_field
+fn eval_nested_field(
+    collection_relationships: &BTreeMap<String, models::Relationship>,
+    variables: &BTreeMap<String, serde_json::Value>,
+    state: &AppState,
+    value: Value,
+    nested_field: &NestedField,
+) -> Result<models::RowFieldValue> {
+    match nested_field {
+        models::NestedField::Object(NestedObject { fields }) => {
+            let full_row: Row = serde_json::from_value(value).map_err(|_| {
+                (
+                    StatusCode::BAD_REQUEST,
+                    Json(models::ErrorResponse {
+                        message: "Expected object".into(),
+                        details: serde_json::Value::Null,
+                    }),
+                )
+            })?;
+            let mut row = IndexMap::new();
+            for (field_name, field) in fields.iter() {
+                row.insert(
+                    field_name,
+                    eval_field(collection_relationships, variables, state, field, &full_row)?,
+                );
+            }
```

Comment:
Is it possible to extract this common function from `execute_query` and reuse it here?

Reply:
Done